connectors/sources/postgresql.py

```diff
@@ -170,127 +399,16 @@ def get_default_configuration(cls):
-
-                        if not rows_length:
-                            break
-
-                        for row in rows:
-                            yield row
-
-                        rows_fetched += rows_length
-                        await self._sleeps.sleep(0)
-                else:
-                    yield cursor.fetchall()  # pyright: ignore
-                break
-            except (InternalClientError, ProgrammingError):
-                raise
-            except Exception as exception:
-                self._logger.warning(
-                    f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
-                )
-                if retry == self.retry_count:
-                    raise exception
-                await self._sleeps.sleep(DEFAULT_WAIT_MULTIPLIER**retry)
-                retry += 1
-
-    async def _async_connect(self, query):
-        """Execute the passed query on the Async supported Database server and return cursor.
-
-        Args:
-            query (str): Database query to be executed.
-
-        Returns:
-            cursor: Asynchronous cursor
-        """
-        try:
-            if self.engine is None:
-                self._create_engine()
-            async with self.engine.connect() as connection:  # pyright: ignore
-                cursor = await connection.execute(text(query))
-                return cursor
-        except Exception as exception:
-            self._logger.warning(
-                f"Something went wrong while executing query. Exception: {exception}"
-            )
-            raise
-
-    def _create_engine(self):
-        """Create async engine for postgresql"""
-        self.engine = create_async_engine(
-            self.connection_string,
-            connect_args=self.get_connect_args(),
-        )
-
-    def get_connect_args(self):
-        """Convert string to pem format and create a SSL context
-
-        Returns:
-            dictionary: Connection arguments
-        """
-        if not self.ssl_enabled:
-            return {}
-
-        pem_format = get_pem_format(key=self.ssl_ca)
-        ctx = ssl.create_default_context()
-        ctx.load_verify_locations(cadata=pem_format)
-        connect_args = {"ssl": ctx}
-        return connect_args
+        self.postgresql_client.close()
 
     async def ping(self):
         """Verify the connection with the database-server configured by user"""
         self._logger.info("Validating the Connector Configuration...")
         try:
-            await anext(
-                self.execute_query(
-                    query=self.queries.ping(),
-                )
-            )
+            await self.postgresql_client.ping()
             self._logger.info("Successfully connected to Postgresql.")
         except Exception as e:
-            raise Exception(f"Can't connect to Postgresql on {self.host}") from e
+            raise Exception("Can't connect to Postgresql.") from e
```

Comment:
Nit: it would still be useful to keep the host in the exception for easier debugging